runtime/src/test/java/com/flipkart/flux/dao/StateMachinesDAOTest.java

```diff
@@ -13,56 +13,87 @@
-    }
+    public void setup() {}
 
-    @Test
-    @Transactional
+    @Test @Transactional
     public void createSMTest() {
-        StateMachinesDAO stateMachinesDAO = injector.getInstance(StateMachinesDAO.class);
-        StatesDAO statesDAO = injector.getInstance(StatesDAO.class);
         DummyTask task = new DummyTask();
         DummyOnEntryHook onEntryHook = new DummyOnEntryHook();
         DummyOnExitHook onExitHook = new DummyOnExitHook();
-        State<DummyEventData> state1 = statesDAO.create(new State<DummyEventData>(2L, "state1", "desc1", onEntryHook, task, onExitHook, 3L, 60L));
-        State<DummyEventData> state2 = statesDAO.create(new State<DummyEventData>(2L, "state2", "desc2", null, null, null, 2L, 50L));
-        Set<State<DummyEventData>> states = new HashSet<State<DummyEventData>>();
+        State<DummyEventData> state1 = statesDAO.create(new State<>(2L, "state1__", "desc1", onEntryHook, task, onExitHook, 3L, 60L));
+        State<DummyEventData> state2 = statesDAO.create(new State<>(2L, "state2__", "desc2", null, null, null, 2L, 50L));
+        Set<State<DummyEventData>> states = new HashSet<>();
         states.add(state1);
         states.add(state2);
-        StateMachine<DummyEventData> stateMachine = new StateMachine<DummyEventData>(2L, "test_name", "test_desc", states);
+        StateMachine<DummyEventData> stateMachine = new StateMachine<>(2L, "SM_name_", "SM_desc_", states);
         String savedSMId = stateMachinesDAO.create(stateMachine).getId();
 
         StateMachine stateMachine1 = stateMachinesDAO.findById(savedSMId);
         Assert.assertNotNull(stateMachine1);
     }
+
+    /** Test creation of State machine from state machine definition */
+    @Test @SuppressWarnings("unchecked")
+    public void createSMDTest() {
+        EventDefinition eventDefinition1 = new EventDefinition("event1");
+        EventDefinition eventDefinition2 = new EventDefinition("event2");
+        Set eventSet1 = new HashSet<>();
+        eventSet1.add(eventDefinition1);
+        Set eventSet2 = new HashSet<>();
+        eventSet2.add(eventDefinition2);
+        StateDefinition stateDefinition1 = new StateDefinition(1L, "state1_", "desc1_", new DummyOnEntryHook(), new DummyTask(), new DummyOnExitHook(), 0L, 60L, eventSet1);
+        StateDefinition stateDefinition2 = new StateDefinition(1L, "state2_", "desc2_", new DummyOnEntryHook(), new DummyTask(), new DummyOnExitHook(), 0L, 60L, eventSet2);
+        Set stateSet = new HashSet<>();
+        stateSet.add(stateDefinition1);
+        stateSet.add(stateDefinition2);
+        StateMachineDefinition stateMachineDefinition = new StateMachineDefinition(1L, "SMD_name_", "SMD_desc_", stateSet);
+
+        String stateMachineId = domainTypeCreator.createStateMachine(stateMachineDefinition).getId();
+
+        StateMachine stateMachine = stateMachinesDAO.findById(stateMachineId);
+        Assert.assertNotNull(stateMachine);
```

Comment:
Sorry. Removed it and comparing equals.

IMO, as equals is using id, and we are using same id to retrieve object, and comparing with old object, checking not null or checking id = id wouldn't make big difference. @yogeshnachnani